tests/lint.rs

```diff
@@ -74,127 +68,32 @@ struct LintRunner {
-                            })
-                            .unwrap();
-                        vec![filename.with_file_name(format!("build_script_build-{}.d", hash))]
-                    }
-                    // target/debug/deps/libforest_wallet-fa26ebcb4b76d710.rmeta
-                    // -> target/debug/deps/forest_wallet-fa26ebcb4b76d710.d
-                    ["bin"] | ["example"] | ["test"] | ["bench"] | ["lib"] => artifact
-                        .filenames
-                        .iter()
-                        .map(|it| {
-                            assert_eq!(it.extension().unwrap(), "rmeta");
-                            let new_file_name = it
-                                .with_extension("d")
-                                .file_name()
-                                .unwrap()
-                                .replacen("lib", "", 1);
-                            it.with_file_name(new_file_name)
-                        })
-                        .collect::<Vec<_>>(),
-                    other => panic!("unexpected artifact.target.kind: {}", other.join(", ")),
-                }
-            })
-            .inspect(|it| debug!(depfile = %it))
-            .map(std::fs::read_to_string)
-            .map(Result::unwrap);
-
-        // 4. Collect all source files by parsing the depfiles
-        let all_source_files = depfiles
-            .flat_map(|depfile| {
-                let dependencies = depfile
-                    .lines()
-                    .filter(|it| !(it.starts_with('#') || it.is_empty()))
-                    .map(|it| {
-                        let (target, _precursors) = it.split_once(':').unwrap();
-                        Utf8PathBuf::from(target)
-                    })
-                    .collect::<Vec<_>>();
-                trace!(dependencies = %dependencies.iter().join(", "));
-                dependencies
-            })
-            .unique();
+        // The initial implementation here tried to ask `cargo` and `rustc`
+        // what the source files were, but it was flaky.
+        //
+        // So just go for a simple globbing of well-known directories.
 
-        // 5. Load all the source files, skipping non-existent or non-rust files
-        let files = all_source_files
-            .flat_map(|path| {
-                let plaintext = std::fs::read_to_string(&path).ok()?;
-                let all = SourceFile::try_from(plaintext).ok()?;
-                Some((path, all))
-            })
-            .collect::<Cache>();
+        let files = [
+            concat!(env!("CARGO_MANIFEST_DIR"), "/src/**/*.rs"),
+            concat!(env!("CARGO_MANIFEST_DIR"), "/tests/**/*.rs"),
+            concat!(env!("CARGO_MANIFEST_DIR"), "/benches/**/*.rs"),
+            concat!(env!("CARGO_MANIFEST_DIR"), "/examples/**/*.rs"),
+        ]
+        .into_iter()
+        .map(glob::glob)
+        .flat_map(Result::unwrap) // PatternError
+        .map(Result::unwrap) // GlobError
```

Comment:
Would using `expect` with, e.g., `PatternError` produce better messages and make the comment unnecessary?